Remove paramiko usage from ipatests

fcami opened the pull request:
Paramiko is not compatible with FIPS.
Migrate all tests using paramiko to the OpenSSH CLI SSH(1).

Fixes: https://pagure.io/freeipa/issue/8129

Note that https://pagure.io/freeipa/issue/8431 was filed as sshpass would need an enhancement to cover all OTP tests.

fcami commented:
hi @mrizwan93 the actual goal of the PR is to remove paramiko usage in our tests.
test_user_permissions is impacted and test_webui will be.
I will refactor to tasks.py once I have a clear view of what's needed. See test_user_permissions.py changes for instance where I use sshpass. The problem is that I need to collect first all auth types we use (pubkey, pass, tgt/keytab) otherwise it'll be a mess. Thanks for the suggestion, that means we agree on where this should go!

fcami commented:
Good runs for [test_commands](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/5240f986-ce69-11ea-ae61-fa163e168469/) and [test_user_permissions](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/5f9259ea-ce69-11ea-b7e9-fa163e5f960e/)

fcami commented:
/AzurePipelines run

fcami commented:
hi @miskopo @mrizwan93 @menonsudhir @tiran Could you please have a look?

miskopo commented on ipatests/pytest_ipa/integration/tasks.py:
```diff
@@ -2297,3 +2298,134 @@ def get_sssd_version(host):
+        )
+
+    ssh_cmd_1 = list(ssh_cmd)
+    if strict_host_key_checking is True:
```
Same as `2412`

fcami replied:
Same reply then

miskopo commented on ipatests/test_integration/test_commands.py:
```diff
@@ -906,45 +891,43 @@ def test_ssh_from_controller(self):
-            client.close()
+            tasks.create_active_user(
+                self.master, test_user, password=password,
+                extra_args=['--homedir', '/root/{}'.format(test_user)]
```
I believe this `homedir` configuration is a mistake

fcami replied:
Yes. Let me fix that. It comes from another PR.

fcami replied:
Fixed and pushed.

fcami commented on ipatests/pytest_ipa/integration/tasks.py:
```diff
@@ -2297,3 +2298,134 @@ def get_sssd_version(host):
+
+    if from_host is not None:
+        raise ValueError(
+            "from_host must be None ; reunning from anywhere but the "
```
Fixed, thank you

fcami commented on ipatests/pytest_ipa/integration/tasks.py:
```diff
@@ -2297,3 +2298,134 @@ def get_sssd_version(host):
+    """
+
+    if from_host is not None:
+        raise ValueError(
```
Yes indeed, fixed.

fcami commented on ipatests/pytest_ipa/integration/tasks.py:
```diff
@@ -2297,3 +2298,134 @@ def get_sssd_version(host):
+        pytest.fail("Unable to run ssh command.", e)
+
+    if auth_method == "password":
+        if expect_auth_success is True:
```
Slightly more, but I don't think it would add anything.

fcami commented on ipatests/pytest_ipa/integration/tasks.py:
```diff
@@ -2297,3 +2298,134 @@ def get_sssd_version(host):
+            assert return_code == 5
+            assert "Authentication succeeded" not in stderr
+    elif auth_method == "key":
+        if expect_auth_success is True:
```
Same reply then

fcami commented on ipatests/pytest_ipa/integration/tasks.py:
```diff
@@ -2297,3 +2298,134 @@ def get_sssd_version(host):
+            assert "Authentication succeeded (publickey)" in stderr
+            # do not assert the return code:
+            # it can be >0 if the command failed.
+        elif expect_auth_failure is True:
```
Same reply then

fcami commented:
@miskopo I think I've addressed your comments except the one about parameter validation. As such, I am not too worried about it as it is clear both variables are expected to be booleans so I'd rather not change the logic there.

fcami commented:
/azp run

miskopo commented:
@fcami Please, quote and remove temp commit and I shall ACK.

fcami commented:
logs:
[test_commands](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/78ae7af4-d0de-11ea-9b8a-fa163e2d2908)
[test_otp](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/d8abc646-d0de-11ea-8b26-fa163e97fbb7)
[test_user_permissions](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/c04080ec-d0de-11ea-b1c5-fa163e07b2f0)
[test_webui_host](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/c9603f64-d0de-11ea-91fe-fa163e54c0e1)
[test_webui_service](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/d21010d0-d0de-11ea-8d8a-fa163e3ad6c5)

fcami commented:
The CI failure in test_commands is unrelated, known, and already fixed in master.
I will rebase and push again.

fcami commented:
/AzurePipelines run

fcami commented:
master:

* d5148c65412d92a7d586cefec66cbabb099fc22f tasks: add run_ssh_cmd
* 73ae4c77f38a7453fa166a01a683c6cd638f2746 ipatests: test_commands: test_ssh_key_connection: Paramiko=>OpenSSH
* 5cc7a2b703b1514112da1d93ee2e3edf24cfb1f7 ipatests: test_user_permissions: test_selinux_user_optimized Paramiko=>OpenSSH
* 27ed8260ba6307fdc22beb10d6f25d13b336ded8 ipatests: test_commands: test_ssh_from_controller: refactor
* 112386f76a4e3bfec9962013ac8ebaadf950ff6f ipatests: test_commands: test_ssh_from_controller: Paramiko=>OpenSSH
* 326e13347c89f2a3287209e95e2b7cd00aa3d80c ipatests: test_commands: test_login_wrong_password: Paramiko=>OpenSSH
* a9f055787ac389e4164f3de5369bade1d4b218b0 ipatests: ui_driver: convert run_cmd_on_ui_host to tasks.py::run_ssh_cmd
* 763d3b059bccc37744764985b57d97554a2a6947 ipatests: test_otp: convert test_2fa_enable_single_prompt to run_ssh_cmd
